src/components/MapboxMap.tsx

```diff
@@ -161,10 +172,79 @@ export function MapboxMap<T>({
   }, [PinComponent, getCoordinate, locationResults]);
 
   return (
-    <div ref={mapContainer} className='h-full w-full' />
+    <div className='grid h-full w-full' ref={parentContainer}>
+      <div ref={mapContainer} className="col-span-full row-span-full invisible"/>
+      {mapboxStaticImageDiv}
+    </div>
   );
 }
 
+function useMapboxStaticImage(
+  mapboxAccessToken: string,
+  parentContainer: React.RefObject<HTMLDivElement>,
+  options: Omit<MapboxOptions, 'container'>,
+): JSX.Element | null {
+  const [divDimension, setDivDimension] = useState<{ width: number, height: number }>();
+  useLayoutEffect(() => {
+    if (parentContainer.current) {
+      setDivDimension({
+        width: parentContainer.current.clientWidth,
+        height: parentContainer.current.clientHeight
+      });
+    }
+  }, [parentContainer]);
+
+  const staticMapboxStyle: React.HTMLAttributes<HTMLDivElement>['style'] = useMemo(() => {
+    const { style, center, zoom } = options;
+    if (!divDimension || !center || !zoom || typeof style !== 'string') {
+      return undefined;
+    }
+    const { width, height } = divDimension;
+    const url = getMapboxStaticImageUrl({ mapboxAccessToken, style, center, zoom, width, height });
+    if (!url) {
+      return undefined;
+    }
+    return {
+      backgroundImage: `url(${url})`
+    };
+  }, [divDimension, mapboxAccessToken, options]);
+
+  if (!staticMapboxStyle) {
+    return null;
+  }
+  return <div id="static-map" className='col-span-full row-span-full' style={staticMapboxStyle}></div>;
+}
+
+/**
+ * The configuration for Mapbox Static Image API url
+ */
+interface MapboxStaticImageUrlConfig {
+  /** Mapbox access token. */
+  mapboxAccessToken: string,
+  /** Mapbox stylesheet url to apply to the static map. */
+  style: string,
+  /** Center point of the static map. */
+  center: LngLatLike,
+  /** Zoom level of the static map. */
+  zoom: number,
+  /** Width (in pixels) of the image. */
+  width: number,
+  /** Height (in pixels) of the image. */
+  height: number
+}
+
+function getMapboxStaticImageUrl(urlConfig: MapboxStaticImageUrlConfig): string | undefined {
+  const { mapboxAccessToken, style, center, zoom, width, height } = urlConfig;
+  // Mapbox Static Image API only support width and height between 1-1280 pixels.
+  if (width > 1280 || height > 1280) {
+    return undefined;
+  }
+  const stylesheet = style.split('mapbox://styles/')[1].split('?')[0];
+  const centerAndZoom = `${center[0]},${center[1]},${zoom}`;
+  const dimension = `${width}x${height}`;
+  return `https://api.mapbox.com/styles/v1/${stylesheet}/static/${centerAndZoom}/${dimension}?access_token=${mapboxAccessToken}`;
+}
```

Comment:
I hate asking for more legal stuff but do we need to poke product about the static image api pricing? https://docs.mapbox.com/api/maps/static-images/#static-images-api-pricing
the first 50k are free which is nice at least https://www.mapbox.com/pricing/#static-images-api

Reply:
per discussion during all hands, tom will run this by product and legal before the feature is merge into develop